pillarbox-player/docs/MediaSession.md

```diff
@@ -0,0 +1,124 @@
+
+- `MediaSession` : `PillarboxMediaSession`
+- `MediaSessionService` : `PillarboxMediaSessionService`
+- `MediaController` : `PillarboxMediaController`
+- `MediaLibrarySession` : `PillarboxMediaLibrarySession`
+- `MediaLibraryService` : `PillarboxMediaLibraryService`
+- `MediaBrowser`: `PillarboxMediaBrowser`
+
+## Connect the player to the `MediaSession`
+
+```kotlin
+val mediaSession = PillarboxMediaSession.Builder(context, player).build()
+```
+
+Remember to release the [`MediaSession`][media-session-documentation] when you no longer need it, or when releasing the player, with:
+
+```kotlin
+mediaSession.release()
+```
+
+More information about [`MediaSession`][media-session-documentation] is available [here][media-session-guide].
+
+## System integration and background playback
+
+AndroidX Media3 library recommends to use [`MediaSessionService`][media-session-service-documentation] or
+[`MediaLibraryService`][media-library-service-documentation] to do background playback. [`MediaLibraryService`][media-library-service-documentation]
+is useful when the application needs to connect to _Android Auto_ or _Automotive_. Pillarbox provides an implementation of each service type to help
+you handle them.
+
+### `PillarboxMediaSessionService`
+
+To use that service, you need to declare it inside your `AndroidManifest.xml`, as follows:
+
+```xml
+<service android:exported="true" android:foregroundServiceType="mediaPlayback" android:name=".service.DemoMediaSessionService">
+    <intent-filter>
+        <action android:name="androidx.media3.session.MediaSessionService" />
+    </intent-filter>
+</service>
+```
+
+And enable foreground service at the top of the manifest:
+
+```xml
+<uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+
+<!-- Only necessary if your target SDK version is 34 or newer -->
+<uses-permission android:name="android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK" />
+```
+
+Then, in your code, you have to use [`PillarboxMediaController`][pillarbox-media-controller-source] to handle playback, instead of
+[`PillarboxExoPlayer`][pillarbox-exo-player-source]. Pillarbox provides an easy way to retrieve the
+[`MediaController`][media-controller-documentation] with [`PillarboxMediaController.Builder`][pillarbox-media-controller-source].
+
+```kotlin
+coroutineScope.launch {
+    val mediaController: PillarboxPlayer = PillarboxMediaController.Builder(context, DemoMediaLibraryService::class.java).build()
+    doSomethingWith(mediaController)
+}
+```
+
+### `PillarboxMediaLibraryService`
+
+[`PillarboxMediaLibraryService`][pillarbox-media-library-service-source] has the same features as
+[`PillarboxMediaSessionService`][pillarbox-media-session-service-source], but it allows the application to provide content with
+[`MediaBrowser`][media-browser-documentation]. More information about [Android auto][android-auto-documentation].
```

Comment:
Not sure which page you wanted to link here for `Android Auto`, but the `[android-auto-documentation]` reference does not exist.